### PR TITLE
V2 more

### PR DESCRIPTION
**Pull request by jianghaolu**

### CODE REVIEW

File: azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+
+@Target(value={METHOD,TYPE})    // The context in which annotation is applicable i.e. this annotation (Host) can be applied to method, method parameter and interface.
+@Retention(RetentionPolicy.RUNTIME)        // Record this annotation in the class file and make it available during runtime.
+public @interface AzureHost {
+    String value() default "";
+    AzureEnvironment.Endpoint enpoint() default Endpoint.RESOURCE_MANAGER;
```
Comment (jianghaolu) [inline azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java]: This is my proposal for multiple environments:

```
@AzureHost(Endpoint.KEY_VAULT)
@Get("secrets/{secretName}")
Secret get(@HostParam String vaultName, @PathParam("secretName") secretName);
```

Comment (daschult) [inline azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java]: Typo: endpoint(), not enpoint()

File: azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.v2.annotations;
```
Comment (daschult) [inline azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java]: I'm a little concerned about putting version numbers into our packages/folder structure. Why not just create a v2 branch off of master that we do all of our work in until we're ready for v2 to be our main version, at which point we would just merge v2 into master and then delete v2?

File: azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+@Target(value={METHOD,TYPE})    // The context in which annotation is applicable i.e. this annotation (Host) can be applied to method, method parameter and interface.
+@Retention(RetentionPolicy.RUNTIME)        // Record this annotation in the class file and make it available during runtime.
+public @interface AzureHost {
+    String value() default "";
+    AzureEnvironment.Endpoint enpoint() default Endpoint.RESOURCE_MANAGER;
+}
+
+/**
+ * The applicability of this annotation is limited to variable with name {host} in the URI template.
```
Comment (daschult) [inline azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java]: These comments are great, well written, and thought-out. Why not make them Javadoc comments on the @interface type itself so that they're available while we're writing code?

File: azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+ *
+ *
+ *  [1] Static annotation applied to interface level.
+ *
+ *    @Host("manage.windowsazure.com")
```
Comment (daschult) [inline azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java]: Why is the interface named "AzureHost" but in the examples it uses "@Host"?

File: client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    private RestProxy(String host, RestClient restClient) {
+        this.host = host;
+        this.restClient = restClient;
+    }
+
+    @Override
+    public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+        if (method.isAnnotationPresent(GET.class)) {
+            String relativePath = method.getAnnotation(GET.class).value();
+            return (host + "/" + relativePath).getBytes();
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java]: Is this supposed to actually make the REST API network call at this point using RxNetty/OkHttp? I'm confused why the proxy is just returned the bytes of the GET URL.

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Contexts.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Retention(RUNTIME)
+@Target(METHOD)
+public @interface Contexts {
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Contexts.java]: What would be an example of a Contexts annotation?

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/DELETE.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.METHOD})            // The context in which annotation is applicable i.e. this annotation (GET) can be applied only to methods
+@Retention(RetentionPolicy.RUNTIME)      // Record this annotation in the class file and make it available during runtime.
+public @interface DELETE {
+    String value() default "";
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/DELETE.java]: Would it be better to not provide a default value so that we don't forget to add the DELETE request's URL?

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/DELETE.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+/**
+ *  For more details on format of "value" field and associated rules refer {@link com.microsoft.rest.Doc_Http_Verb_Annotation_Value_URI_Template}
+ *
+ *  Below shows an example where 'DELETE' annotation is applied to the method with relative path uri template:
+ *
+ *  @GET("subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}")
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/DELETE.java]: Update example to use a DELETE annotation instead of GET.

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Doc_Http_Verb_Annotation_Value_URI_Template.java [hunk 9]
```diff
@@ -36,7 +36,7 @@
  *  The name of these parameters are predefined, if they are present then they must be present exactly in places where they are suppose to present in a valid URI.
  *
  *  Value of {protocol} is derived from {@link EndpointProtocol} annotation. Refer documentation of {@link EndpointProtocol} for more details.
- *  Value of {host} is derived from {@link EndpointHost} annotation. Refer documentation of {@link EndpointHost} for more details.
+ *  Value of {host} is derived from {@link Host} annotation. Refer documentation of {@link Host} for more details.
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Doc_Http_Verb_Annotation_Value_URI_Template.java]: Why did you change this from EndpointHost to Host when the other two values have an "Endpoint" prefix?

File: azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+import com.microsoft.azure.AzureEnvironment.Endpoint;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+
+@Target(value={METHOD,TYPE})    // The context in which annotation is applicable i.e. this annotation (Host) can be applied to method, method parameter and interface.
```
Comment (daschult) [inline azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java]: EndpointHost could be targeted to a PARAMETER. Is there a reason you don't allow AzureHost to target a PARAMETER?

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/HEAD.java [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+public @interface HEAD {
+    String value() default "";
+}
+
+/**
+ *  For more details on format of "value" field and associated rules refer {@link com.microsoft.rest.Doc_Http_Verb_Annotation_Value_URI_Template}
+ *
+ *  Below shows an example where 'HEAD' annotation is applied to the method with relative path uri template:
+ *
+ *  @GET("subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}")
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/HEAD.java]: Update example to use HEAD.

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/HeaderParam.java [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Retention(RUNTIME)
+@Target(PARAMETER)
+public @interface HeaderParam {
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/HeaderParam.java]: Can you add an example for this annotation?

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Headers.java [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Retention(RUNTIME)
+@Target(METHOD)
+public @interface Headers {
+    /**
+     * List of static headers.
+     */
+    String[] value();
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Headers.java]: Do these strings take the format "headerName=headerValue1,headerValue2"? Does that mean that the annotation parser will have to parse the list of headers?

File: azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java (hunk 1 above)
Comment (daschult) [inline azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java]: Typo: endpoint(), not enpoint()

File: azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java (hunk 4 above)
Comment (daschult) [inline azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java]: Can you change these examples to use AzureHost instead of Host?

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Host.java [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+
+@Target(value={METHOD,TYPE})    // The context in which annotation is applicable i.e. this annotation (Host) can be applied to method, method parameter and interface.
+@Retention(RetentionPolicy.RUNTIME)        // Record this annotation in the class file and make it available during runtime.
+public @interface Host {
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Host.java]: Why do we allow both AzureHost and Host? What is the benefit of using one over the other?

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/PATCH.java [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+package com.microsoft.rest.v2.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.METHOD})            // The context in which annotation is applicable i.e. this annotation (GET) can be applied only to methods
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/PATCH.java]: Change comment annotation reference from (GET) to (PATCH).

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/PATCH.java [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+public @interface PATCH {
+    String value() default "";
+}
+
+/**
+ *  For more details on format of "value" field and associated rules refer {@link com.microsoft.rest.Doc_Http_Verb_Annotation_Value_URI_Template}
+ *
+ *  Below shows an example where 'PATCH' annotation is applied to the method with relative path uri template:
+ *
+ *  @GET("subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}")
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/PATCH.java]: Update example to use @PATCH annotation instead of @GET.

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/POST.java [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+package com.microsoft.rest.v2.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.METHOD})            // The context in which annotation is applicable i.e. this annotation (GET) can be applied only to methods
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/POST.java]: Update comment reference to use (POST) instead of (GET).

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/POST.java [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+public @interface POST {
+    String value() default "";
+}
+
+/**
+ *  For more details on format of "value" field and associated rules refer {@link com.microsoft.rest.Doc_Http_Verb_Annotation_Value_URI_Template}
+ *
+ *  Below shows an example where 'POST' annotation is applied to the method with relative path uri template:
+ *
+ *  @GET("subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}")
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/POST.java]: Update example to use @POST instead of @GET.

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/PUT.java [hunk 19]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+package com.microsoft.rest.v2.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.METHOD})            // The context in which annotation is applicable i.e. this annotation (GET) can be applied only to methods
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/PUT.java]: Update comment reference to be (PUT) instead of (GET).

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/PUT.java [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+@Retention(RetentionPolicy.RUNTIME)      // Record this annotation in the class file and make it available during runtime.
+public @interface PUT {
+    String value() default "";
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/PUT.java]: Each of the HTTP Method annotations have a field named "value", even though value is actually the URL path. Is there a reason we didn't name it "path"?

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/PUT.java [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+/**
+ *  For more details on format of "value" field and associated rules refer {@link com.microsoft.rest.Doc_Http_Verb_Annotation_Value_URI_Template}
+ *
+ *  Below shows an example where 'PUT' annotation is applied to the method with relative path uri template:
+ *
+ *  @GET("subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}")
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/PUT.java]: Update example to use PUT.

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/QueryParam.java [hunk 22]
```diff
@@ -0,0 +1,28 @@
...
+     * The name of the variable in the endpoint uri template which will be replaced with the value
+     * of the parameter annotated with this annotation.
+     */
+    String value();
+    /**
+     * A value true for this argument indicates that value of {@link QueryParam#value()} is already encoded
+     * hence engine should not encode it, by default value will be encoded.
+     */
+    boolean encoded() default false;
+}
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/QueryParam.java]: Can you provide an example of using this annotation?

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/QueryParam.java [hunk 23]
```diff
@@ -0,0 +1,28 @@
...
+public @interface QueryParam {
+    /**
+     * The name of the variable in the endpoint uri template which will be replaced with the value
+     * of the parameter annotated with this annotation.
+     */
+    String value();
+    /**
+     * A value true for this argument indicates that value of {@link QueryParam#value()} is already encoded
+     * hence engine should not encode it, by default value will be encoded.
+     */
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/QueryParam.java]: Consider this: "This value indicates whether or not this parameter's value is encoded. If the parameter's value is encoded, the engine will not attempt to encode it again. The default value is false, meaning the engine will attempt to encode the parameter's value."

File: azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java (hunk 2 above)
Comment (jianghaolu) [inline azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java]: The convention in Java world is to add version numbers to package names so that multiple versions can be used together

File: azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java (hunk 10 above)
Comment (jianghaolu) [inline azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java]: Because this is not supported by Swagger 2.0 - to hit a completely different endpoint (parametrized ones are allowed) you need a separate swagger doc anyways

File: azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java (hunk 3 above)
Comment (jianghaolu) [inline azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java]: I think we started out here because these are not actually finalized javadocs? @anuchandy I agree they will go into javadocs largely

File: client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java (hunk 5 above)
Comment (jianghaolu) [inline client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java]: This is a just a place holder - > this is where your work will do the magic :)

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Contexts.java (hunk 6 above)
Comment (jianghaolu) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Contexts.java]: logging context.
Or anything that you'd like to pass down the pipeline

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/DELETE.java (hunk 7 above)
Comment (jianghaolu) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/DELETE.java]: I think this is the same for @GET which is allowing @Url parameters to specify the entire path

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Doc_Http_Verb_Annotation_Value_URI_Template.java (hunk 9 above)
Comment (jianghaolu) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Doc_Http_Verb_Annotation_Value_URI_Template.java]: I think the other 2 should go away

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Headers.java (hunk 13 above)
Comment (jianghaolu) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Headers.java]: Yes and yes

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Host.java (hunk 14 above)
Comment (jianghaolu) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Host.java]: The `AzureEnvironment.Endpoint` enum isn't available in the base `client-runtime`

File: azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java (hunk 2 above)
Comment (daschult) [inline azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java]: Fair enough.

File: client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java (hunk 5 above)
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java]: That's what I thought after I looked at the unit test at the end of the review.

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Doc_Http_Verb_Annotation_Value_URI_Template.java (hunk 9 above)
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Doc_Http_Verb_Annotation_Value_URI_Template.java]: The prefix to the other two annotations should go away, or the other two annotations should go away?

File: azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java (hunk 3 above)
Comment (anuchandy) [inline azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java]: yes, i put it there just like that, once we finalize we can move them to javadoc section.

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Contexts.java (hunk 6 above)
Comment (anuchandy) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Contexts.java]: Should this be a map instead of array of strings? so that we can easily lookup.

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/DELETE.java (hunk 7 above)
Comment (anuchandy) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/DELETE.java]: and if both `@url` and `value()` are empty, then it would be a get on the base url right? or an error?

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/PUT.java (hunk 20 above)
Comment (anuchandy) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/PUT.java]: One reason i can think of is, the value can contain not just "path segments" but query params, fragments along with path segments.

Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/PUT.java]: That's a good point. In that case, I'd say we should have a Javadoc comment explaining what could go into value, or we should rename it to pathQueryFragment. (I prefer the Javadoc comment.)

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Contexts.java (hunk 6 above)
Comment (anuchandy) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/Contexts.java]: ahh ignore my comment, its statically defined.

File: client-runtime/src/main/java/com/microsoft/rest/v2/annotations/HeaderParam.java [hunk 24]
```diff
@@ -12,6 +12,29 @@
...
+ *
+ * Example 1:
+ *   {@literal @}PUT("{functionId}")
+ *   Observable<Response<ResponseBody>> createOrReplace(@PathParam("functionId", encoded = true) String functionId, @BodyParam FunctionInner function, @HeaderParam("If-Match") String ifMatch);
+ *
+ *   "If-Match: user passed value" will show up as one of the headers.
+ *
+ * Example 2:
+ *   {@literal @}GET("subscriptions/{subscriptionId}/providers/Microsoft.ServiceBus/namespaces")
+ *   Observable<Response<ResponseBody>> list(@Path("subscriptionId") String subscriptionId, @Header("accept-language") String acceptLanguage, @Header("User-Agent") String userAgent);
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/HeaderParam.java]: @HeaderParam, right?

Comment (jianghaolu) [inline client-runtime/src/main/java/com/microsoft/rest/v2/annotations/HeaderParam.java]: Right.
